pkg/api/norman/customization/aks/listers.go

```diff
@@ -373,27 +373,87 @@ func listClusters(ctx context.Context, cap *Capabilities) ([]byte, int, error) {
 	return encodeOutput(clusters)
 }
 
+const (
+	AzureSkuResourceTypeVM            = "virtualMachines"
+	AzureAcceleratedNetworkingFeature = "AcceleratedNetworkingEnabled"
+)
+
 func listVMSizes(ctx context.Context, cap *Capabilities) ([]byte, int, error) {
 	if cap.ResourceLocation == "" {
 		return nil, http.StatusBadRequest, fmt.Errorf("region is required")
 	}
 
-	virtualMachine, err := NewVirtualMachineClient(cap)
+	type VMSizeInfo struct {
+		//Name is synonymous with the size shown in Rancher UI
+		Name                           string
+		AcceleratedNetworkingSupported bool
+		AvailabilityZones              []string
+	}
+
+	skuClient, err := NewVirtualMachineSKUClient(cap)
 	if err != nil {
 		return nil, http.StatusInternalServerError, err
 	}
-	vmMachineSizeList, err := virtualMachine.List(ctx, cap.ResourceLocation)
+
+	req, err := skuClient.ListPreparer(ctx)
+	if err != nil {
+		return nil, http.StatusInternalServerError, err
+	}
+
+	// only get resources for the given location.
+	// this is the only filter supported by API version 2017-09-01,
+	// which is the latest version at time of writing.
+	// A guide to azure API filtering can be found here
+	// https://learn.microsoft.com/en-us/rest/api/monitor/filter-syntax
+	q := req.URL.Query()
+	q.Add("$filter", fmt.Sprintf("location eq '%s'", cap.ResourceLocation))
+	req.URL.RawQuery = q.Encode()
+
+	resp, err := skuClient.ListSender(req)
+	if err != nil {
+		return nil, resp.StatusCode, err
+	}
+
+	skuList, err := skuClient.ListResponder(resp)
 	if err != nil {
-		return nil, http.StatusBadRequest, fmt.Errorf("failed to get VM sizes: %v", err)
+		return nil, http.StatusInternalServerError, fmt.Errorf("failed to get VM sizes: %v", err)
 	}
 
-	vmSizes := make([]string, 0, len(*vmMachineSizeList.Value))
+	if skuList.Value == nil {
+		return nil, http.StatusNotFound, err
+	}
+
+	var VMSkuInfo []VMSizeInfo
+	for _, resourceSku := range *skuList.Value {
+		// we currently can't specify a particular SKU type in the API request,
+		// we have to filter them out here
+		if to.String(resourceSku.ResourceType) != AzureSkuResourceTypeVM {
+			continue
+		}
+
+		vm := VMSizeInfo{
+			Name: to.String(resourceSku.Name),
+		}
+
+		if resourceSku.Capabilities != nil {
+			for _, skuCapabilities := range *resourceSku.Capabilities {
+				if to.String(skuCapabilities.Name) == AzureAcceleratedNetworkingFeature && to.String(skuCapabilities.Value) == "True" {
+					vm.AcceleratedNetworkingSupported = true
+					break
+				}
+			}
+		}
+
+		if resourceSku.LocationInfo != nil && len(*resourceSku.LocationInfo) > 0 {
```

Comment:
Seems like these objects use a lot of pointers to maps/slices, i.e. double pointers. Is it just part of the azure sdk? Nothing that can be addressed?

Reply:
Yeah. A lot of cloud provider sdks written in go use pointers heavily, so it's nothing we can change on our end.